### PR TITLE
KOGITO-2207 - Create process without adding to ProcessInstanceManager

### PR DESCRIPTION
**Pull request by cristianonicolai**
…hen loading process

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](CONTRIBUTING.md)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: jbpm/jbpm-flow/src/main/java/org/kie/kogito/process/impl/AbstractProcessInstance.java [hunk 1]
```diff
@@ -125,6 +125,7 @@ public void internalSetProcessInstance(org.kie.api.runtime.process.ProcessInstan
         if (this.legacyProcessInstance != null && this.status != ProcessInstance.STATE_PENDING) {
             throw new IllegalStateException("Impossible to override process instance that already exists");
         }
+        ((WorkflowProcessInstanceImpl) this.legacyProcessInstance).disconnect();
```
Comment (mswiderski) [inline jbpm/jbpm-flow/src/main/java/org/kie/kogito/process/impl/AbstractProcessInstance.java:128]: why is this needed? Since process instance was loaded from the storage it should not be connected at all. So how does it get into connected state here as the reconnect is few lines below?

Comment (cristianonicolai) [reply jbpm/jbpm-flow/src/main/java/org/kie/kogito/process/impl/AbstractProcessInstance.java:128]: yep, I'm doing this only so that legacy process instance is removed from the _DefaultProcessInstanceManager_, otherwise it will keep appending more and more instances to it.

Comment (mswiderski) [reply jbpm/jbpm-flow/src/main/java/org/kie/kogito/process/impl/AbstractProcessInstance.java:128]: why is it added there in the first place?

Comment (cristianonicolai) [reply jbpm/jbpm-flow/src/main/java/org/kie/kogito/process/impl/AbstractProcessInstance.java:128]: This is done when unmarshalling the process: https://github.com/kiegroup/kogito-runtimes/blob/master/jbpm/jbpm-flow/src/main/java/org/kie/kogito/process/impl/marshalling/ProcessInstanceMarshaller.java#L102
and https://github.com/kiegroup/kogito-runtimes/blob/master/jbpm/jbpm-flow/src/main/java/org/kie/kogito/process/impl/AbstractProcessInstance.java#L112

Comment (mswiderski) [reply jbpm/jbpm-flow/src/main/java/org/kie/kogito/process/impl/AbstractProcessInstance.java:128]: in that case I'd change it there as it looks very confusing to disconnect and then reconnect.

Comment (cristianonicolai) [reply jbpm/jbpm-flow/src/main/java/org/kie/kogito/process/impl/AbstractProcessInstance.java]: @mswiderski ok so trying a new approach here, so create process instance does not add it to _ProcessInstanceManager_. Some side effects are that now id is only generated when process is started and not created. Same goes for validating business key, id can only be done when starting the process now, not when creating. wdyt?

**Comment by cristianonicolai** [comment]
jenkins please retest this

File: jbpm/jbpm-flow/src/main/java/org/kie/kogito/process/impl/AbstractProcessInstance.java [hunk 2]
```diff
@@ -84,7 +75,7 @@
 
     protected Integer status;
     protected String id;
-    protected String businessKey;
+    protected CorrelationKey correlationKey;
```
Comment (mswiderski) [inline jbpm/jbpm-flow/src/main/java/org/kie/kogito/process/impl/AbstractProcessInstance.java:78]: why this change is needed? correlation key type is way to complex and kind of legacy too, so keeping string here and use correlation key on legacy process instance would make more sense to me, but not a huge issue I guess.

Comment (cristianonicolai) [reply jbpm/jbpm-flow/src/main/java/org/kie/kogito/process/impl/AbstractProcessInstance.java:78]: @mswiderski I end up changing this because the the `CorrelationKey` object is needed for when adding the process instance to the `ProcessInstanceManager`. I can revert that but then I would need to recreate this object every time the process is started. See: https://github.com/kiegroup/kogito-runtimes/pull/582/files#diff-c72f318d542b5bbf6d5de58b1b5c2f1fR163

**Comment by cristianonicolai** [comment]
jenkins please retest this

File: jbpm/jbpm-flow/src/main/java/org/kie/kogito/process/impl/AbstractProcessInstance.java (hunk 2 above)
Comment (MarianMacik) [reply jbpm/jbpm-flow/src/main/java/org/kie/kogito/process/impl/AbstractProcessInstance.java:78]: @cristianonicolai Just a clarification. Being this a process instance, if we want to start two process instances, then we need to first create 2 process instance objects anyway, right? That would mean that the recreation will only be needed for the second time when the process instance is started (so once it will be done in the constructor, second time in the star method). But to me it makes sense to compute this in the constructor and then reuse it in whatever method within the same object.

Comment (cristianonicolai) [reply jbpm/jbpm-flow/src/main/java/org/kie/kogito/process/impl/AbstractProcessInstance.java:78]: @MarianMacik correct, we need to first create then start, so it makes sense to me to be done at the constructor level.
